Actions/CheckForUpdates/CheckForUpdates.ps1

```diff
@@ -204,6 +204,14 @@ if ($update -ne 'Y') {
 else {
     # $update set, update the files
     try {
+        if (!$directCommit) {
+            $env:GH_TOKEN = $token
+            $existingPullRequest = (gh api --paginate /repos/$env:GITHUB_REPOSITORY/pulls -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" | ConvertFrom-Json) | Where-Object { $_.title -eq "Update AL-Go System Files - $templateSha" -and $_.base.ref -eq $updateBranch } | Select-Object -First 1
```

Comment:
To optimize a little, consider filtering on the base branch as part of the API call.